test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/HtmlHelperDisplayExtensionsTest.cs

```diff
@@ -0,0 +1,413 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.TestCommon;
+using Microsoft.AspNetCore.Mvc.ViewEngines;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Rendering
+{
+    public class HtmlHelperDisplayExtensionsTest
+    {
+        [Fact]
+        public void DisplayHelpers_FindsModel_WhenViewDataIsNotSet()
+        {
+            // Arrange
+            var expected = $"<div class=\"HtmlEncode[[display-label]]\">HtmlEncode[[SomeProperty]]</div>{Environment.NewLine}" + 
+                $"<div class=\"HtmlEncode[[display-field]]\"></div>{Environment.NewLine}";
+            var model = new SomeModel();
+            var viewEngine = new Mock<ICompositeViewEngine>(MockBehavior.Strict);
+            viewEngine
+                .Setup(v => v.GetView(/*executingFilePath*/ null, It.IsAny<string>(), /*isMainPage*/ false))
+                .Returns(ViewEngineResult.NotFound(string.Empty, Enumerable.Empty<string>()));
+            viewEngine
+                .Setup(v => v.FindView(It.IsAny<ActionContext>(), It.IsAny<string>(), /*isMainPage*/ false))
+                .Returns(ViewEngineResult.NotFound(string.Empty, Enumerable.Empty<string>()));
+            var helper = DefaultTemplatesUtilities.GetHtmlHelper(model, viewEngine.Object);
+
+            // Act
+            var displayResult = helper.Display(expression: string.Empty);
+            var displayNullResult = helper.Display(expression: null);   // null is another alias for current model
+            var displayForResult = helper.DisplayFor(m => m);
+            var displayForModelResult = helper.DisplayForModel();
+
+            // Assert
+            Assert.Equal(expected, HtmlContentUtilities.HtmlContentToString(displayResult));
+            Assert.Equal(expected, HtmlContentUtilities.HtmlContentToString(displayNullResult));
+            Assert.Equal(expected, HtmlContentUtilities.HtmlContentToString(displayForResult));
+            Assert.Equal(expected, HtmlContentUtilities.HtmlContentToString(displayForModelResult));
+        }
+
+        [Fact]
+        public void Display_UsesAdditionalViewData()
+        {
+            // Arrange
+            var model = new SomeModel { SomeProperty = "ModelValue" };
+            var view = new Mock<IView>();
+            view.Setup(v => v.RenderAsync(It.IsAny<ViewContext>()))
+                .Callback(async (ViewContext v) =>
+                {
+                    await v.Writer.WriteAsync(v.ViewData["SomeProperty"].ToString());
```

Comment:
Clean these lambdas up by removing `async`, `await`, and `ToString()`. You don't need curly braces in most of them either.